### PR TITLE
only parse the text of an incoming USSD message if it was provided.

### LINKED ISSUE #464: long USSD code returns "UNKNOWN response. Please report the error" in smsd
I have encountered the following problem with SMSD and long USSD codes:

After USSD code is sent by gammu-smsd a first response from network is received. The first response is corrrectly received and saved in inbox. The first response is followed by a second response for USSD, which is not recognized by smsd. An error is thrown: _gammu: UNKNOWN response. Please report the error, see <https://wammu.eu/support/bugs/>. Thank you_ . As a result sent USSD code stays in outbox and is scheduled for a retry. 
The above process is retried in loop x-times.

Here is an excerpt from the log:

```
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: UNKNOWN response. Please report the error, see <https://wammu.eu/support/bugs/>. Thank you
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: LAST SENT frame type 0x00/length 108
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 41A|54T|2B+|43C|55U|53S|44D|3D=|311|2C,|22"|300|300|322|41A|300 AT+CUSD=1,"002A0
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 300|333|377|300|300|333|377|300|300|333|322|300|300|322|41A|300 03700370032002A0
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 300|333|355|300|300|333|311|300|300|333|377|300|300|333|322|300 0350031003700320
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 300|333|366|300|300|333|355|300|300|333|366|300|300|333|388|300 0360035003600380
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 300|322|41A|300|300|333|311|300|300|333|322|300|300|333|355|300 02A0031003200350
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 300|322|41A|300|300|333|311|300|300|333|399|300|300|333|399|300 02A0031003900390
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 300|333|344|300|300|322|333|22"|2C,|311|355|0D                  0340023",15.    
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: RECEIVED frame type 0x00/length 0xa/10
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 2B+|43C|55U|53S|44D|3A:|20 |322|0D |0A                          +CUSD: 2..      
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: 
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: GSM_DialService failed with error UNKNOWNRESPONSE[16]: Unknown response from phone. See <https://wammu.eu/support/bugs/> for information on how to report it.
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: gammu: Leaving GSM_DialService
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: Error sending SMS: Unknown response from phone. See <https://wammu.eu/support/bugs/> for information on how to report it. (UNKNOWNRESPONSE[16])
Thu 2019/04/11 09:12:11 gammu-smsd[1028]: Execute SQL: UPDATE outbox SET "SendingTimeOut" = now() + interval '30 seconds', "Retries" = 1 WHERE "ID" = '19681';

```

Here is a full log: 
[USSD_unknown_response.txt](https://github.com/gammu/gammu/files/3080161/USSD_unknown_response.txt)

### PR DESCRIPTION
**Pull request by kstuart**
Doesn't attempt to parse +CUSD parameters beyond __code__ when no other parameters were provided.

Fixes #464

### CODE REVIEW

File: tests/atgen/smsd-incoming-ussd.c [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    // AT+CUSD=1
+    "OK\r\n",
+    // AT+CSCS="UCS2"
+    "OK\r\n",
+    // AT+CSCS?
+    "+CSCS:\r",
+    "OK\r\n",
+    // USSD DialService
+    "+CUSD: 0,\"0076006F00740072006500200072006500710075006500740065002000650073007400200065006E00200063006F0075007200730020006400650020007400720061006900740065006D0065006E0074002E00200076006500750069006C006C0065007A0020006C0069007200650020006C00650020006D00650073007300610067006500200070006C0075007300200074006100720064002E\",15\r\n",
+    "OK\r\n",
```
Comment (nijel) [inline tests/atgen/smsd-incoming-ussd.c]: Can you please also add testcase for the response only with code?

Comment (kstuart) [reply tests/atgen/smsd-incoming-ussd.c]: Both cases are tested in the same function, line 20 is the incoming __+CUSD__ with only a code,  the test that is was received without error is lines 57-64. (lines 53-55 test the __+CUSD__ with all arguments)

The test is done like that during diagnostics to follow as closely as possible the pattern in the log attached to the issue, where a USSD request was made and 2 incoming USSD responses were received, the second of which was __+CUSD: 2__. I can extract it into its own testcase of you prefer?

Comment (nijel) [reply tests/atgen/smsd-incoming-ussd.c]: I don't mind having the test for complete response chain, it just didn't include the `+CUSD: 2` line, which is what you've fixed in the code.

**Comment by nijel** [comment]
I've restarted the AppVeyor build, it seems to crashed on their side without any clear reason.

File: tests/atgen/smsd-incoming-ussd.c (hunk 1 above)
Comment (kstuart) [reply tests/atgen/smsd-incoming-ussd.c]: The test was there just easy to miss,  [line 20](https://github.com/gammu/gammu/blob/0bbafb481ead963ac00aa272119eed1612a29da6/tests/atgen/smsd-incoming-ussd.c#L20) declared the code only event and it was tested on [lines 57-64](https://github.com/gammu/gammu/blob/0bbafb481ead963ac00aa272119eed1612a29da6/tests/atgen/smsd-incoming-ussd.c#L57-L64).

It was a bit late and I decided it was fine to leave the debugging test, but it's a lot clearer now the test just addresses the change.

**Comment by kstuart** [comment]
Thanks for giving it a kick, AppVeyor was struggling last night a bit I think, looks all good now though.

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**